### PR TITLE
Add support for `$ne` operator

### PR DESCRIPTION
**Pull request by ribaraka**
Refs #251.

### CODE REVIEW

**Comment by AlekSi** [comment]
After #467

**Review by seeforschauer — APPROVED** [review]

File: integration/query_comparision_test.go [hunk 1]
```diff
@@ -944,8 +944,166 @@ func TestQueryComparisonLte(t *testing.T) {
...
+				Name:    "BadValue",
+				Message: `Can't have regex as arg to $ne.`,
+			},
+		},
+	} {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			var actual []bson.D
```
Comment (seeforschauer) [inline integration/query_comparision_test.go:1095]: here we use to move actual to the line 1101, and add before that line a blank one.

**Review by AlekSi — COMMENTED** [review]
Let's finish #471 first, then add documents and arrays to that PR

**Review by AlekSi — COMMENTED** [review]
Please merge with `main` and resolve merge conflicts.

And did we forgot tests for `$in`?

**Comment by ribaraka** [comment]
> Please merge with `main` and resolve merge conflicts.
> 
> And did we forgot tests for `$in`?

Ok.
There was no PR for $in operator yet.

**Review by seeforschauer — APPROVED** [review]

File: integration/query_comparision_test.go [hunk 2]
```diff
@@ -1074,6 +1074,190 @@ func TestQueryComparisonNin(t *testing.T) {
...
 
-// $in
+func TestQueryComparisonNe(t *testing.T) {
+	t.Parallel()
+	providers := []shareddata.Provider{shareddata.Scalars, shareddata.Composites}
+	ctx, collection := setup(t, providers...)
+
+	for name, tc := range map[string]struct {
+		value         any
+		valueToBeLost string
```
Comment (AlekSi) [inline integration/query_comparision_test.go:1084]: That's ID, not a value. A better name might be `unexpectedID`

Comment (ribaraka) [reply integration/query_comparision_test.go]: fixed

**Review by AlekSi — APPROVED** [review]

**Pull request merged**